analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/MetricsAnalyzerTest.cs

```diff
@@ -87,6 +87,25 @@ public class MetricsAnalyzerTest
                         // There should be no metrics messages for the cshtml files.
                         messages.Select(x => Path.GetFileName(x.FilePath)).Should().BeEquivalentTo("_Imports.razor"));
 
+        [TestMethod]
+        public void VerifyMetrics_CSharp12() =>
+            CreateBuilder(false, CSharp12FileName)
+                .WithAdditionalFilePath(AnalysisScaffolding.CreateSonarProjectConfig(TestContext, ProjectType.Product))
+                .VerifyUtilityAnalyzer<MetricsInfo>(messages =>
+                {
+                    messages.Should().ContainSingle();
+                    var metrics = messages.Single();
+                    metrics.ClassCount.Should().Be(1); // no changes
+                    metrics.CodeLine.Should().HaveCount(12);
+                    metrics.CognitiveComplexity.Should().Be(1); // no changes
+                    metrics.Complexity.Should().Be(3); // no changes
+                    metrics.ExecutableLines.Should().HaveCount(3); // 5, 7, 9
+                    metrics.FunctionCount.Should().Be(2); // no changes
```

Comment:
Should the count of functions rather be 3? The constructor `PrimaryConstructor(bool condition, int n)` is included in the count, so should `PrimaryConstructor(bool condition /* comment */)`, I guess.
To be checked with the other team members and PM, to determine whether we should create an issue, in the upcoming sprint, to fix the "Function Count" metric.

Reply:
That's a good point, not sure what would be the correct behavior, but I lean towards considering it a function, for the reason you mentioned. I asked on the slack channel 👍